src/processing/strategies/transform.rs

```diff
@@ -0,0 +1,38 @@
+use crate::processing::strategies::{CommitRequest, MessageRejected, ProcessingStrategy};
+use crate::types::Message;
+
+pub struct Transform<TPayload: Clone + Send + Sync, TTransformed: Clone + Send + Sync> {
+    pub function: fn(Message<TPayload>) -> TTransformed,
```

Comment:
Can the function also return an error if it fails to run the transformation?

Reply:
Updated this return a `Result<TTransformed, InvalidMessage>`. We're not actually handling the error in any way yet though as I haven't thought about how a DLQ would fit in here yet.